app/js/mongodb.js

```diff
@@ -2,18 +2,21 @@ const Settings = require('settings-sharelatex')
 const { MongoClient, ObjectId } = require('mongodb')
 
 const clientPromise = MongoClient.connect(Settings.mongo.url)
-const dbPromise = clientPromise.then((client) => client.db())
-
-async function getCollection(name) {
-  return (await dbPromise).collection(name)
-}
 
 async function waitForDb() {
   await clientPromise
 }
 
+const db = {}
+waitForDb().then(async function () {
+  const internalDb = (await clientPromise).db()
+
+  db.messages = internalDb.collection('messages')
```

Comment:
What do you think about making this slightly dynamic? I think it might be a small improvement to say something like

```JavaScript
const COLLECTIONS = ['messages', 'rooms']
// ...
waitForDb().then(async function () {
  const internalDb = (await clientPromise).db()

  for (const collection of COLLECTIONS) {
    db[collection] = internalDb.collection(collection)
  }
}
```

I think it makes it slightly more obvious what's happening, and aids reuse in other apps?

Reply:
I do not have a strong opinion on that. I liked that my IDE could auto-complete the `db` attribute name, but this was not possible with mongojs either, so it's not a show-stopper.

With a fully dynamic version I think it would make sense to move all of it into an npm package and pull the list of collections from the settings (yes, not ideal, but we _abuse_ them for other things already -- like redis-key functions --, so this should be fine).

```js
assert(require('settings-sharelatex').mongo.collections.join() === ['messages','rooms'].join())
const { db, waitForDb } = require('@overleaf/mongo')

waitForDb().then(async function () {
  const message = await db.message.findOne({...})
  ...
}) 
```

Reply:
cc @emcsween ^

Reply:
> I liked that my IDE could auto-complete the db attribute name

That's interesting. Maybe an argument for making it less magical. For web it will be a relatively long list, but for most services I imagine it will be pretty short.

Reply:
I don't have a strong opinion, but I slightly prefer the explicit setting:
```js
db.messages = internalDb.collection('messages')
...
```
I think that it's as easy to read as the alternatives, and helping the IDE and/or the eventual type checker figuring out the shape of what's exported is worth the tiny code duplication.

On the other hand, my least preferred alternative is putting the list of collections in the settings. This opens the door to someone overriding the settings with something else, which would never be correct since there is a single list of collections that the code needs.

Reply:
Alright, I'll leave it as is then.

Reply:
Works for me. :+1: